grammar/haxe.bnf

```diff
@@ -254,11 +254,12 @@ private topLevelDeclaration ::= classDeclaration
 
 privateKeyWord ::= 'private'
 externOrPrivate ::= 'extern' | privateKeyWord
+private externAndMaybePrivate ::= privateKeyWord? 'extern' privateKeyWord?
 
 typedefDeclaration ::= macroClassList? externOrPrivate? 'typedef' componentName genericParam? '=' functionTypeWrapper ';'?
 {pin=5 mixin="com.intellij.plugins.haxe.lang.psi.impl.AbstractHaxeTypeDefImpl" implements="com.intellij.plugins.haxe.lang.psi.HaxeClass"}
 
-externClassDeclaration ::= macroClassList? externOrPrivate* ('class' | 'interface') componentName genericParam? inheritList? '{' externClassDeclarationBody '}'
+externClassDeclaration ::= macroClassList? externAndMaybePrivate ('class' | 'interface') componentName genericParam? inheritList? '{' externClassDeclarationBody '}'
```

Comment:
Maybe I'm getting a bit confused here...
The way it is now, "extern interface" and "extern class" are both externClassDeclarations, and "interface" without the extern is an interfaceDeclaration.  I would think that all interfaces should be interfaceDeclarations, and classes should be classDeclarations.

But that aside, isn't the point of this change to make interfaces not be extern?  I guess I have to go do some more reading.